src/pharmpy/tools/mfl/filter.py

```diff
@@ -23,4 +25,23 @@
 )
 
 
-structsearch_statement_types = (DirectEffect, EffectComp, IndirectEffect)
+structsearch_statement_types = (DirectEffect, EffectComp, IndirectEffect, Metabolite)
+
+
+def mfl_filtering(statements, tool_name=""):
```

Comment:
This was done since the algorithm was built for and is located within the modelsearch tool. I feel like the default value could still be `modelsearch` in order to be consistent with the other algorithms which are lacking this argument.
Since only `exhaustive_stepwise` is used in structsearch, it feels unnecessary to implement the same argument for the other algorithms. By not having `modelsearch` as default, this would indicate a special case when running the modelsearch tool which, if the `exhaustive_stepwise` algorithm is used, would require the specification of tool_name="modelsearch". I feel like that is maybe unnecessary for the moment, but would love your input :)

Reply:
I think it's more that I find it a bit confusing that there is a default that will raise an `AssertionError` if not provided. To me it makes more sense to not have a default here at all if it's not clear which tool would be a sensible default.

Reply:
I realize i was talking about another function (exhaustive_stepwise) which also got an update with the same argument (tool_name).
I realize this was in regard to the filtering operation and fully agree with you now. Will remove the default value! Sorry about the confusion